### ISSUE TITLE
Incorrect parameter type for NSG ids for OKE Native pod networking

### ISSUE THREAD

**Issue by hyder** [oracle/terraform-provider-oci#1662]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

When creating an OKE node pool, if Native Pod Networking is used, the pod nsg id needs to be passed as a parameter. Currently, the following is expected:

```
        node_pool_pod_network_option_details {
            #Required
            cni_type = var.node_pool_node_config_details_node_pool_pod_network_option_details_cni_type

            #Optional
            max_pods_per_node = var.node_pool_node_config_details_node_pool_pod_network_option_details_max_pods_per_node
            pod_nsg_ids = var.node_pool_node_config_details_node_pool_pod_network_option_details_pod_nsg_ids
            pod_subnet_ids = var.node_pool_node_config_details_node_pool_pod_network_option_details_pod_subnet_ids
        }
```

`pod_nsg_ids` is named in the plural leading developers to pass a collection e.g. list. However, a string is expected by this parameter and therefore only 1 NSG id can be passed.

If the intention is to allow several NSGs to be used, then this parameter should accept a list of NSG ids.

If the intention is to allow only 1 NSG to be used, then this parameter should be renamed to its singular form `pod_nsg_id` and keep accepting a string as is currently the case.

### Affected Resource(s)

`oci_containerengine_node_pool`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. 
# Please remove any sensitive information from configuration files before sharing them. 
```
https://raw.githubusercontent.com/oracle-terraform-modules/terraform-oci-oke/main/modules/oke/nodepools.tf
### Debug Output
```
│   on modules/oke/nodepools.tf line 47, in resource "oci_containerengine_node_pool" "nodepools":
│   47:         pod_nsg_ids       = var.pod_nsgs
│     ├────────────────
│     │ var.pod_nsgs is list of tuple with 1 element
│
│ Inappropriate value for attribute "pod_nsg_ids": incorrect list element type: string required.
```

### Steps to Reproduce

1. `terraform apply`

**Comment by ravinitp** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by XinruXiao-9** [comment]
Thank you for reporting the issue. Our intention is to allow multiple NSGs to be used, so pod_nsg_ids is accepting a list of NSG ids. From the error log, you are using a list of tuple instead of list of string. Please change var.pod_nsgs to list of NSG ids and retry. Thanks! 

example: 
```
variable "pod_nsgs" {
  default = ["ocid1.networksecuritygroup.oc1.iad.aaaaaaaadcnekdgrpkjb7rnumm4fl2zfbzd7kwuavreusg7hduavkpdpl33a","ocid1.networksecuritygroup.oc1.iad.aaaaaaaahzrpc6sivisdwt5yrjr4k5rfo56deucqjqizio4kv2ju3uzcv4eq"]
}
```

**Comment by tf-oci-pub** [comment]
Fixed in version 4.123.0 by @XinruXiao-9
